lib/src/barback/dartdevc/module.dart

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:barback/barback.dart';
+
+/// Serializable object that describes a single `module`.
+///
+/// A `module` is a set of dart [AssetId]s contained in the module and a list of
+/// other [AssetId]s which are direct dependencies.
+///
+/// Note that all [assetIds] must be in the same package, but
+/// [directDependencies] may be from any package.
+///
+/// Use a [ModuleReader#readTransitiveDeps] to get the transitive dependencies
+/// of a module.
+class Module {
+  final ModuleId id;
+  final Set<AssetId> assetIds;
+  final Set<AssetId> directDependencies;
+
+  Module(this.id, this.assetIds, this.directDependencies);
+  Module.fromJson(List<List<dynamic>> json)
```

Comment:
I added a description of the parts relevant to this class - some of it is left up to `ModuleId.toJson` and `AssetId.serialize`.